maybe-uninitialized warnings under cc()

`R CMD {build,INSTALL}` don't throw these, but `cc()` does, at least on our Codespace:

```
idatetime.c: In function ‘convertDate’:
idatetime.c:149:13: warning: ‘ctype’ may be used uninitialized [-Wmaybe-uninitialized]
  149 |             convertSingleDate(ix[i], ctype, &ansp[i]);
      |             ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
idatetime.c:131:14: note: ‘ctype’ was declared here
  131 |     datetype ctype;
      |              ^~~~~
frollR.c: In function ‘frollfunR’:
frollR.c:182:11: warning: ‘ialign’ may be used uninitialized [-Wmaybe-uninitialized]
  182 |   #pragma omp parallel for if (ialgo==0) schedule(dynamic) collapse(2) num_threads(getDTthreads(nx*nk, false))
      |           ^~~
frollR.c:102:7: note: ‘ialign’ was declared here
  102 |   int ialign;                                                   // decode align to integer
      |       ^~~~~~
frollR.c:182:11: warning: ‘sfun’ may be used uninitialized [-Wmaybe-uninitialized]
  182 |   #pragma omp parallel for if (ialgo==0) schedule(dynamic) collapse(2) num_threads(getDTthreads(nx*nk, false))
      |           ^~~
frollR.c:136:20: note: ‘sfun’ was declared here
  136 |   enum {MEAN, SUM} sfun;
      |                    ^~~~
frollR.c:177:8: warning: ‘ialgo’ may be used uninitialized [-Wmaybe-uninitialized]
  177 |     if (ialgo==0)
      |        ^
frollR.c:159:16: note: ‘ialgo’ was declared here
  159 |   unsigned int ialgo;                                           // decode algo to integer
      |                ^~~~~
frollR.c: In function ‘frollapplyR’:
frollR.c:292:7: warning: ‘ialign’ may be used uninitialized [-Wmaybe-uninitialized]
  292 |       frollapply(dx[i], inx[i], dw, ik[j], &dans[i*nk+j], ialign, dfill, pc, rho, verbose);
      |       ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
frollR.c:247:7: note: ‘ialign’ was declared here
  247 |   int ialign;
      |       ^~~~~~
nafill.c: In function ‘nafillR’:
nafill.c:198:11: warning: ‘itype’ may be used uninitialized [-Wmaybe-uninitialized]
  198 |   #pragma omp parallel for if (nx>1) num_threads(getDTthreads(nx, true))
      |           ^~~
nafill.c:167:16: note: ‘itype’ was declared here
  167 |   unsigned int itype;
      |                ^~~~~
fifelse.c: In function ‘fcaseR’:
fifelse.c:394:54: warning: ‘pthens’ may be used uninitialized [-Wmaybe-uninitialized]
  394 |           if (!naout) SET_VECTOR_ELT(ans, idx, pthens[idx & thenMask]);
      |                                                      ^
fifelse.c:389:28: note: ‘pthens’ was declared here
  389 |       const SEXP *restrict pthens;
      |                            ^~~~~~
fifelse.c:377:56: warning: ‘pthens’ may be used uninitialized [-Wmaybe-uninitialized]
  377 |           SET_STRING_ELT(ans, idx, naout ? pna : pthens[idx & thenMask]);
      |                                                        ^
fifelse.c:371:28: note: ‘pthens’ was declared here
  371 |       const SEXP *restrict pthens;
      |                            ^~~~~~
```

Removed all but the `-Wmaybe-uninitialized` issues here for clarity. Opening a second issue for the other one we should address.

--- a/src/fifelse.c
+++ b/src/fifelse.c
@@ -368,7 +368,7 @@ SEXP fcaseR(SEXP rho, SEXP args) {
       }
     } break;
     case STRSXP: {
-      const SEXP *restrict pthens;
+      const SEXP *restrict pthens=NULL;
       if (!naout) pthens = STRING_PTR_RO(thens); // the content is not useful if out is NA_LOGICAL scalar
       const SEXP pna = NA_STRING;
       for (int64_t j=0; j<len2; ++j) {
@@ -386,7 +386,7 @@ SEXP fcaseR(SEXP rho, SEXP args) {
     case VECSXP: {
       // the default value of VECSXP is `NULL` so we don't need to explicitly
       // assign the NA values as it does for other atomic types
-      const SEXP *restrict pthens;
+      const SEXP *restrict pthens=NULL;
       if (!naout) pthens = SEXPPTR_RO(thens); // the content is not useful if out is NA_LOGICAL scalar
       for (int64_t j=0; j<len2; ++j) {
         const int64_t idx = imask ? j : p[j];

--- a/src/frollR.c
+++ b/src/frollR.c
@@ -99,7 +99,7 @@ SEXP frollfunR(SEXP fun, SEXP obj, SEXP k, SEXP fill, SEXP algo, SEXP align, SEX
   if (LOGICAL(hasna)[0]==FALSE && LOGICAL(narm)[0])
     error(_("using hasNA FALSE and na.rm TRUE does not make sense, if you know there are NA values use hasNA TRUE, otherwise leave it as default NA"));
 
-  int ialign;                                                   // decode align to integer
+  int ialign=-2;                                                   // decode align to integer
   if (!strcmp(CHAR(STRING_ELT(align, 0)), "right"))
     ialign = 1;
   else if (!strcmp(CHAR(STRING_ELT(align, 0)), "center"))
@@ -133,7 +133,7 @@ SEXP frollfunR(SEXP fun, SEXP obj, SEXP k, SEXP fill, SEXP algo, SEXP align, SEX
     dx[i] = REAL(VECTOR_ELT(x, i));                             // assign source columns to C pointers
   }
 
-  enum {MEAN, SUM} sfun;
+  enum {MEAN, SUM} sfun=0;
   if (!strcmp(CHAR(STRING_ELT(fun, 0)), "mean")) {
     sfun = MEAN;
   } else if (!strcmp(CHAR(STRING_ELT(fun, 0)), "sum")) {
@@ -156,7 +156,7 @@ SEXP frollfunR(SEXP fun, SEXP obj, SEXP k, SEXP fill, SEXP algo, SEXP align, SEX
     LOGICAL(hasna)[0]==TRUE ? 1 :                               // hasna TRUE, might be some NAs
     -1;                                                         // hasna FALSE, there should be no NAs
 
-  unsigned int ialgo;                                           // decode algo to integer
+  unsigned int ialgo=-1;                                           // decode algo to integer
   if (!strcmp(CHAR(STRING_ELT(algo, 0)), "fast"))
     ialgo = 0;                                                  // fast = 0
   else if (!strcmp(CHAR(STRING_ELT(algo, 0)), "exact"))
@@ -244,7 +244,7 @@ SEXP frollapplyR(SEXP fun, SEXP obj, SEXP k, SEXP fill, SEXP align, SEXP rho) {
     error(_("n must be non 0 length"));
   int *ik = INTEGER(k);
 
-  int ialign;
+  int ialign=-2;
   if (!strcmp(CHAR(STRING_ELT(align, 0)), "right")) {
     ialign = 1;
   } else if (!strcmp(CHAR(STRING_ELT(align, 0)), "center")) {

--- a/src/idatetime.c
+++ b/src/idatetime.c
@@ -128,7 +128,7 @@ SEXP convertDate(SEXP x, SEXP type)
     const int n = length(x);
     if (!isString(type) || length(type) != 1)
         internal_error(__func__, "invalid type for, should have been caught before"); // # nocov
-    datetype ctype;
+    datetype ctype=0;
     bool ansint = true;
     if (!strcmp(CHAR(STRING_ELT(type, 0)), "yday")) ctype = YDAY;
     else if (!strcmp(CHAR(STRING_ELT(type, 0)), "wday")) ctype = WDAY;

--- a/src/nafill.c
+++ b/src/nafill.c
@@ -164,7 +164,7 @@ SEXP nafillR(SEXP obj, SEXP type, SEXP fill, SEXP nan_is_na_arg, SEXP inplace, S
     }
   }
 
-  unsigned int itype;
+  unsigned int itype=-1;
   if (!strcmp(CHAR(STRING_ELT(type, 0)), "const"))
     itype = 0;
   else if (!strcmp(CHAR(STRING_ELT(type, 0)), "locf"))

